src/base/kaldi-error.cc

```diff
@@ -18,156 +19,122 @@
 // limitations under the License.
 
 #ifdef HAVE_EXECINFO_H
-#include <execinfo.h>  // To get stack trace in error messages.
-// If this #include fails there is an error in the Makefile, it does not
-// support your platform well. Make sure HAVE_EXECINFO_H is undefined, and the
-// code will compile.
-#ifdef HAVE_CXXABI_H
-#include <cxxabi.h>  // For name demangling.
-// Useful to decode the stack trace, but only used if we have execinfo.h
-#endif  // HAVE_CXXABI_H
+  #include <execinfo.h>  // To get stack trace in error messages.
+  // If this #include fails there is an error in the Makefile, it does not
+  // support your platform well. Make sure HAVE_EXECINFO_H is undefined, 
+  // and the code will compile.
+  #ifdef HAVE_CXXABI_H
+    #include <cxxabi.h>  // For name demangling.
```

Comment:
It's not specifically forbidden in that text, but I've never seen code like
this elsewhere in Google stuff or in Kaldi-- please put it back.

On Tue, May 10, 2016 at 1:42 PM, Karel Vesely <notifications@github.com>
wrote:

> In src/base/kaldi-error.cc
> <https://github.com/kaldi-asr/kaldi/pull/755#discussion_r62717661>:
>
> > @@ -18,156 +19,122 @@
> >  // limitations under the License.
> >
> >  #ifdef HAVE_EXECINFO_H
> > -#include <execinfo.h>  // To get stack trace in error messages.
> > -// If this #include fails there is an error in the Makefile, it does not
> > -// support your platform well. Make sure HAVE_EXECINFO_H is undefined, and the
> > -// code will compile.
> > -#ifdef HAVE_CXXABI_H
> > -#include <cxxabi.h>  // For name demangling.
> > -// Useful to decode the stack trace, but only used if we have execinfo.h
> > -#endif  // HAVE_CXXABI_H
> > +  #include <execinfo.h>  // To get stack trace in error messages.
> > +  // If this #include fails there is an error in the Makefile, it does not
> > +  // support your platform well. Make sure HAVE_EXECINFO_H is undefined,
> > +  // and the code will compile.
> > +  #ifdef HAVE_CXXABI_H
> > +    #include <cxxabi.h>  // For name demangling.
>
> Okay. Well, it is not strictly against the Google style:
> https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros
>
> Indents make the macros more readable, but I can put it back as it was if
> it is an issue...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/755/files/24bef8dcb4c501a59324debcafd9dcd426bfde23#r62717661>
>